### PR TITLE
PIN-4332 - Update of daily calls  with the same value of the previous version is not allowed

### PR DESCRIPTION
**Pull request by nttdata-rtorsoli**

### CODE REVIEW

**Review by galales — CHANGES_REQUESTED** [review]
this feature is not related to `updatePurpose` and `updateReversePurpose` paths, but to `createPurposeVersion`.
Updating a draft purpose is always allowed, the issue arises when creating a new version that has the same dailyCalls as the previous

File: src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala [hunk 1]
```diff
@@ -734,6 +735,15 @@ final case class PurposeApiServiceImpl(
     else Future.failed(PurposeNotInDraftState(purpose.id))
   }
 
+  private def assertDailyCallsIsDifferentThanBefore(purpose: PersistentPurpose, dailyCalls: Int): Future[Unit] = {
+    val ordering: Ordering[OffsetDateTime] = Ordering(Ordering.by[OffsetDateTime, Long](_.toEpochSecond).reverse)
+    val previousDailyCalls                 = purpose.versions.sortBy(_.createdAt)(ordering).map(_.dailyCalls).headOption
+    previousDailyCalls match {
+      case Some(x) if x != dailyCalls => Future.successful(())
+      case _                          => Future.failed(DailyCallsEqualThanBefore(purpose.id))
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala]: albeit similar, with this change we cover exactly the case we need (while leaving the absence of versions a responsibility of other checks)
```suggestion
      case Some(x) if x == dailyCalls => Future.failed(DailyCallsEqualThanBefore(purpose.id))
      case _                          => Future.successful(())
```

File: src/main/scala/it/pagopa/interop/purposeprocess/error/PurposeProcessErrors.scala [hunk 2]
```diff
@@ -91,5 +91,6 @@ object PurposeProcessErrors {
       extends ComponentError("0026", s"EService ${eServiceId.toString} has not Receive mode")
   final case class RiskAnalysisNotFound(eServiceId: UUID, riskAnalysisId: UUID)
       extends ComponentError("0027", s"EService $eServiceId does not contain Risk Analysis $riskAnalysisId")
-
+  final case class DailyCallsEqualThanBefore(purposeId: UUID)
+      extends ComponentError("0028", s"Purpose $purposeId has the same dailyCalls")
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/purposeprocess/error/PurposeProcessErrors.scala]: minor readability improvement
```suggestion
  final case class UnchangedDailyCalls(purposeId: UUID)
      extends ComponentError("0028", s"Creation of new version without changing daily calls for purpose $purposeId")
```

**Review by galales — COMMENTED** [review]
minor points

**Review by galales — APPROVED** [review]
🚀

**Review by beetlecrunch — APPROVED** [review]

**Pull request merged**